pkg/dataplane/http/context.go

```diff
@@ -1549,3 +1549,48 @@ func trimAndParseInt(str string) (int, error) {
 	trimmed := strings.TrimSpace(str)
 	return strconv.Atoi(trimmed)
 }
+
+// PutOosObject
+func (c *context) PutOosObject(putOosObjectInput *v3io.PutOosObjectInput,
+	context interface{},
+	responseChan chan *v3io.Response) (*v3io.Request, error) {
+	return c.sendRequestToWorker(putOosObjectInput, context, responseChan)
+}
+
+// PutOosObjectSync
+func (c *context) PutOosObjectSync(putOosObjectInput *v3io.PutOosObjectInput) error {
+
+	// concatenate header + data lengths with ',' separator
```

Comment:
1. Prefer range over for with an index
2. Formatting + concatenating is more CPU intensive than using StringBuilder (strings are immutable, you are copying to a new string each time). Also prefer strconv over Sprintf
3. Look for helper objects in the standard library over doing things like copying arrays with offsets. In this case I think bytes.Buffer can do the trick with no overhead

Suggest the following (untested):
```
var iovecSizes strings.Builder

totalSize := len(putOosObjectInput.Header)
iovecSizes.WriteString(strconv.Itoa(headerLength))

for _, ioVec := range putOosObjectInput.Data {
	totalSize += len(ioVec)
	iovecSizes.WriteString(",")
	iovecSizes.WriteString(strconv.Itoa(len(ioVec)))
}

buffer := bytes.NewBuffer(make([]byte, 0, totalSize))
buffer.Write(putOosObjectInput.Header)

for _, ioVec := range putOosObjectInput.Data {
	buffer.Write(ioVec)
}

putOosObjectHeaders["slice"] = strconv.Itoa(putOosObjectInput.SliceId)
putOosObjectHeaders["io-vec-num"] = strconv.Itoa(len(putOosObjectInput.Data) + 1)
putOosObjectHeaders["io-vec-sizes"] = iovecSizes.Bytes()
```

Reply:
You can also use a heuristic to Grow the stringbuilder to reduce allocations